src/app/data/page.tsx

```diff
@@ -189,104 +338,8 @@ function DataContent() {
-                    </dd>
-                  </div>
-                  <div className="flex items-center justify-between">
-                    <dt className="text-muted-foreground">Fees</dt>
-                    <dd>
-                      {transaction &&
-                        `${formattedFees} ${selectedChain?.ticker}`}
-                    </dd>
-                  </div>
-                  <div className="flex items-center justify-between">
-                    <dt className="text-muted-foreground">Gas</dt>
-                    <dd>{transaction?.parsed?.gas}</dd>
-                  </div>
-                  {/* TODO Handle multiple senders */}
-                  <div className="flex items-center justify-between">
-                    <dt className="text-muted-foreground">Sender</dt>
-                    <dd>
-                      {transaction?.parsed?.senders?.length &&
-                        transaction?.parsed?.senders[0].address}
-                    </dd>
-                  </div>
-                  {/* TODO Handle multiple recipients */}
-                  {transaction?.parsed?.recipients?.length && (
-                    <div className="flex items-center justify-between">
-                      <dt className="text-muted-foreground">Recipient</dt>
-                      <dd>{transaction?.parsed?.recipients[0].address}</dd>
-                    </div>
-                  )}
-                  {/* TODO Move to a specific "staking" section ? */}
-                  {transaction?.parsed?.validators?.source && (
-                    <div className="flex items-center justify-between">
-                      <dt className="text-muted-foreground">
-                        Source validator
-                      </dt>
-                      <dd>{transaction?.parsed?.validators.source.address}</dd>
-                    </div>
-                  )}
-                  {/* TODO Move to a specific "staking" section ? */}
-                  {transaction?.parsed?.validators?.target && (
-                    <div className="flex items-center justify-between">
-                      <dt className="text-muted-foreground">
-                        Target validator
-                      </dt>
-                      <dd>{transaction?.parsed?.validators.target.address}</dd>
-                    </div>
-                  )}
-                  <div className="flex items-center justify-between">
-                    <dt className="text-muted-foreground">Nonce</dt>
-                    <dd>{transaction?.parsed?.nonce}</dd>
-                  </div>
-                  <div className="flex items-center justify-between">
-                    <dt className="text-muted-foreground">Memo</dt>
-                    <dd>{transaction?.parsed?.memo}</dd>
-                  </div>
-                </dl>
-              )}
-            </div>
+          <CardContent className="max-h-[50vh] overflow-y-auto">
+            {renderParsedData(transaction)}
```

Comment:
U will want some loader here to have any feedback when we click button otherwise user will find it weird that nothing happened when he click the loop